### PR TITLE
Torq5 savedown bugfix

### PR DESCRIPTION
**Pull request by CReid98**
Fixing a bug where savedown was erroring when called on an empty table

- added error trapping when savedown is called
- fixed issue with undefined variables
- adjusted formatting to improve readability

### CODE REVIEW

File: code/wdb/datastripe.q [hunk 1]
```diff
@@ -60,25 +58,30 @@ if[.ds.datastripe;.proc.addinitlist[(`initdatastripe;`)]];
...
+    directory:` sv .Q.par[dir;partitionint;tablename],`;
+    .lg.o[`save;"Saving ",string[s]," data from ",string[tablename]," table to partition ",string[partitionint],". Table contains ",string[count enumdata]," rows."];
+    .lg.o[`save;"Saving data down to ",string[directory]];
+
+    /- upsert select data matched on partition to specific directory
+    .[upsert;
+        (directory;enumdata);
+        {[e] .lg.e[`upserttopartition;"Failed to save table to disk : ",e];'e}
+    ];
+    };
```
Comment (jonathonmcmurray) [inline code/wdb/datastripe.q:84]: with the indentation change, it's difficult to see what if anything actually changed here?

File: code/rdb/datastripe.q [hunk 2]
```diff
@@ -2,13 +2,15 @@
...
+    t:tables[`.] except .rdb.ignorelist;
+
     // clear data from tables
     lasttime:nextp-.ds.periodstokeep*(nextp-currp);
     tabs:.ds.deletetablebefore'[t;`time;lasttime];
     .lg.o[`reload;"Kept ",string[.ds.periodstokeep]," period",$[.ds.periodstokeep>1;"s";""]," of data from : ",", " sv string[tabs]];
 
     // update the access table in the rdb
-    .rdb.access:update start:lasttime^(.ds.getstarttime each key .rdb.tablekeycols), stptime:data[][`time] from .rdb.access;
+    .rdb.access:update start:lasttime^(.ds.getstarttime each key .rdb.tablekeycols), stptime:data[][`time] from .rdb.access;
```
Comment (jonathonmcmurray) [inline code/rdb/datastripe.q:13]: remove this whitespace change

**Review by jonathonmcmurray — COMMENTED** [review]
Can we get some context in the PR description? What was the bug and how does this change fix it?

File: code/wdb/datastripe.q (hunk 1 above)
Comment (CReid98) [reply code/wdb/datastripe.q:84]: no changes made here, just changing indentation and spacing to try and improve readability

File: code/rdb/datastripe.q (hunk 2 above)
Comment (CReid98) [reply code/rdb/datastripe.q]: whitespace has been removed

File: code/wdb/datastripe.q (hunk 1 above)
Comment (CReid98) [reply code/wdb/datastripe.q:84]: Ok, I'll keep this in mind going forward

File: code/rdb/datastripe.q [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
     .lg.o[`reload;"reload command has been called remotely"];
 
+    t:tables[`.] except .rdb.ignorelist;
```
Comment (jonathonmcmurray) [inline code/rdb/datastripe.q:5]: so `t` wasn't defined before?

File: code/wdb/datastripe.q [hunk 4]
```diff
@@ -27,9 +27,7 @@ modaccess:{[accesstab]};
 
     // call the savedown function
     .ds.savealltablesoverperiod[.ds.td;nextp;lasttime];
-
-    tabs:.ds.deletetablebefore'[t;`time;lasttime];
-    .lg.o[`reload;"Kept ",string[.ds.periodstokeep]," period",$[.ds.periodstokeep>1;"s";""]," of data from : ",", " sv string[tabs]];
+    .lg.o[`reload;"Kept ",string[.ds.periodstokeep]," period",$[.ds.periodstokeep>1;"s";""]," of data from : ",", " sv string[t]];
```
[the remark this answers is not in the document]
Comment (CReid98) [reply code/wdb/datastripe.q:30]: This function is also called inside the .ds.savetablesoverperiod function, meaning this line is redundant

File: code/rdb/datastripe.q (hunk 3 above)
Comment (CReid98) [reply code/rdb/datastripe.q:5]: No, this was an oversight as I had assumed it was defined when the initdatastripe function was run

**Review by jonathonmcmurray — APPROVED** [review]

**Pull request merged**